Std/Tactic/Omega/Frontend.lean

```diff
@@ -80,6 +80,18 @@ def mkAtomLinearCombo (e : Expr) : OmegaM (LinearCombo × OmegaM Expr × HashSet
 local macro_rules
   | `(mkAppN $f #[$xs,*]) => (xs.getElems.foldlM (fun x e => `(Expr.app $x $e)) f : MacroM Term)
 
+
+/-- Match on the two ways of spelling successor that are defeq : `n+1`, `n.succ`.
+TODO: support b^(1+n), add a theorem to rewrite b^(1+n) = b^n*b. -/
+def succ? (e : Expr) : Option Expr :=
+  match e.getAppFnArgs with
+  | (``Nat.succ, #[n]) => some n
+  | (``HAdd.hAdd, #[_, _, _, _, a, b]) => do
+     if b == toExpr (1 : Nat)
```

Comment:
What does `==` on expressions check? I would assume it's only syntactic equality, which seems potentially over-specific: what if for some reason the `OfNat (Int)` instance is syntactically different in a user's file?
We could either do the match, where we allow the instance to be whatever, or we should check for defEq. (Can we use `MetaM` inside of `OmegaM`?)

Reply:
Yes, it's syntactic. I would be terribly surprised if the `ofNat` instance is different, and I picked up this technique from the Omega file that builds atoms.

Reply:
Sure, it seems I might be a bit too paranoid :sweat_smile: